docs/database/supplementary-guidelines/migration-guide.md

```diff
@@ -0,0 +1,249 @@
+affected by them, existing database instrumentations published by
+OpenTelemetry are required to implement a migration plan that will assist users in
+transitioning to the stable database semantic conventions.
+
+Specifically, when existing database instrumentations published by OpenTelemetry are
+updated to the stable database semantic conventions, they:
+
+- SHOULD introduce an environment variable `OTEL_SEMCONV_STABILITY_OPT_IN` in
+  their existing major version, which accepts:
+  - `database` - emit the stable database conventions, and stop emitting
+    the old database conventions that the instrumentation emitted previously.
+  - `database/dup` - emit both the old and the stable database conventions,
+    allowing for a phased rollout of the stable semantic conventions.
+  - The default behavior (in the absence of one of these values) is to continue
+    emitting whatever version of the old database conventions the
+    instrumentation was emitting previously.
+- Need to maintain (security patching at a minimum) their existing major version
+  for at least six months after it starts emitting both sets of conventions.
+- May drop the environment variable in their next major version and emit only
+  the stable database conventions.
+
+## Summary of changes
+
+This section summarizes the changes made to the HTTP semantic conventions
+from
+[v1.24.0](https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/database/README.md).
+to
+[v1.TODO (stable)](https://github.com/open-telemetry/semantic-conventions/blob/v1.TODO/docs/database/README.md).
+
+### Database client span attributes
+
+<!-- prettier-ignore-start -->
+| Change                                              | Comments                                                                                                    |
+|-----------------------------------------------------|-------------------------------------------------------------------------------------------------------------|
+| `db.connection_string`                              | Removed                                                                                                     |
+| `db.user`                                           | Removed                                                                                                     |
+| `network.transport`                                 | Removed                                                                                                     |
+| `network.type`                                      | Removed                                                                                                     |
+| `db.name` &rarr; `db.namespace`                     |                                                                                                             |
+| `db.redis.database_index` &rarr; `db.namespace`     |                                                                                                             |
```

Comment:
`db.redis.database_index` is `long` data type. whereas `db.namespace` is a `string` type. 

should we mark `db.redis.database_index` as removed instead? or can we add some comments to explain the cross-data type replacement?

Reply:
Redis lettuce API `getDatabase` is returning the Redis database number. What value is expected for the new db.namespace for reids? 

https://lettuce.io/core/release/api/io/lettuce/core/RedisURI.html#PARAMETER_NAME_DATABASE:~:text=public%C2%A0int-,getDatabase,-()

Reply:
`redisUri.getHost()` as namespace?

Reply:
I just sent #1233 to clarify, let me know if that helps